specs/Standards/Tokens/FungibleTokenMetadata.md

```diff
@@ -1,60 +1,45 @@
+## Reference-level explanation
 
-Prior art:
+A fungible token contract implementing the metadata standard shall contain a field named `ft_metadata`. The field will link to a structure with the properties from the interface below.
 
-- [EIP-1046](https://eips.ethereum.org/EIPS/eip-1046)
-- [OpenZeppelin's ERC-721 Metadata standard](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#ERC721Metadata) also helped, although it's for non-fungible tokens.
+**Interface**:
 
-## Guide-level explanation
+```ts
+type FungibleTokenMetadata = {
+    spec: string;
+    name: string;
+    symbol: string;
+    defaultIcon: bytes|null;
+    darkIcon: bytes|null;
+    reference: string|null;
+    reference_hash: string|null;
+    decimals: number;
+}
+```
 
-A fungible token smart contract allows for discoverable properties. Some properties can be determined by other contracts on-chain, or return in view method calls. Others can only be determined by an oracle system to be used on-chain, or by a frontend with the ability to access a linked reference file.
+# Discussion
 
-### Examples scenario
+## On-chain vs. offchain hosted data
 
-#### Token provides metadata upon deploy and initialization
+Currently deployment on a NEAR token smart contract is 200 kb. The cost of 5 kb metadata on the cost of code deployment on the top of the contract deployment is miniscule. Icons are in the range of 5 kBytes - 200 kBytes per icon, so they are comparable to a smart contract deployment. 
 
-Alice deploys a wBTC fungible token contract.
+Off-chain data comes complicated with privacy and tracking issues as pointed out by Robert Zarembe.
 
-**Assumptions**
+NEAR is the best blockchain in the world. For the reducing complexity, and after discussing with people on Discord, my stance is that everything on-chain. NEAR should scale for this as it scales for many things.
 
-- The wBTC token contract is `wbtc`.
-- Alice's account is `alice`.
-- The precision ("decimals" in this metadata standard) on wBTC contract is `10^8`.
+## Icon format
 
-**High-level explanation**
+### Light theme vs. dark theme
 
-Alice issues a transaction to deploy and initialize the fungible token contract, providing arguments to the initialization function that set metadata fields.
+Two different icons needed: `defaultIcon` (black text on white background) for iOS, websites. `darkIcon` or websites supporting dark theme and Android UI. After discussing with Ethereum wallet vendors, there are multiple cases where icon looks good on one platform, but horrible on another, because of color bleed issues on the stroke lines of an icon. Token projects are not happy if their icon looks like amaeur.
 
-**Technical calls**
+Thus, we should provide both light and dark icons by default.
 
-1. `alice` deploys a contract and calls `wbtc::new` with all metadata. If this deploy and initialization were done using [NEAR CLI](https://docs.near.org/docs/tools/near-cli) the command would be:
+### SVG vs. PNG
```

Comment:
again, I don't think we should standarize it (themes, file format, dimensions ....), especially on-chain. I would prefer to have a _Recommendation_ section where we put all that considerations. This is not needed for consensus. What we want to recommend is best practice in terms of usability and censorship resistance.